src/controllers/Screen/displayPage/displayTasks.js

```diff
@@ -33,23 +35,33 @@ const displayTasks = (project) => {
 };
 
 const generateTaskDiv = (task) => {
+  const app = createAppController();
+
   const taskId = task.getId();
   const taskDetails = task.viewDetails();
 
   const article = document.createElement("article");
   article.classList.add("task");
   article.setAttribute("data-id", taskId);
-  article.addEventListener("mousedown", () => displayTaskDetailsInModal(task));
+  article.addEventListener("click", (e) => {
+    displayTaskDetailsInModal(task);
+  });
   if (taskDetails.urgency) article.classList.add("urgent");
   if (taskDetails.completed) article.classList.add("completed");
 
   // left-side
   const checkboxDiv = document.createElement("div");
   checkboxDiv.classList.add("checkbox-container");
+  checkboxDiv.addEventListener("click", (e) => e.stopPropagation());
```

Comment:
it was very interesting that i had to do this here instead of in the checkbox element below (which didn't work - checking the checkbox just ended up firing the article parent event handler, which opened the modal showing details of the task)